setup.py

```diff
@@ -65,7 +65,10 @@
 }
 
 if sys.version_info < (3, 8):
-    extras_require['tests'] += ["hdf5 ==1.12.1"]  # To be able to solve on mamba
+    extras_require['tests'] += [
+        "fsspec ==2023.1",  # Bad conda-forge fsspec 2023.3.0 release
+        "hdf5 ==1.12.1",  # To be able to solve on mamba
```

Comment:
The PyPI build is currently failing as `hdf5` is not a Python package available on PyPI.